From 90d1e77ad5e51d60a8fa35a0e826286744a58181 Mon Sep 17 00:00:00 2001
Message-Id: <patch-1.1-90d1e77ad5e-20211217T001606Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Wed, 15 Dec 2021 09:14:08 +0100
Subject: [PATCH] Makefile: remove -DPAGER_ENV from BASIC_CFLAGS
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Remove -DPAGER_ENV from the BASIC_CFLAGS and instead have it passed
via the EXTRA_CPPFLAGS passed when compiling pager.c.

This doesn't change anything except to make it clear that only pager.c
needs this, as it's the only user of this define. See
995bc22d7f8 (pager: move pager-specific setup into the build,
2016-08-04) for the commit that originally added this.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
Range-diff:
 1:  1621ca72c1d <  -:  ----------- Makefile: don't invoke msgfmt with --statistics
 2:  b7c36c9fea0 <  -:  ----------- Makefile: don't set up "perl/build" rules under NO_PERL=Y
 3:  29b000eb0f1 <  -:  ----------- Makefile: use "=" not ":=" for po/* and perl/*
 4:  daead5ec293 <  -:  ----------- Makefile: clean perl/build/ even with NO_PERL=Y
 5:  3c987590740 <  -:  ----------- Makefile: remove "mv $@ $@+" dance redundant to .DELETE_ON_ERROR
 6:  b57f582ccd3 <  -:  ----------- Makefile: guard Perl-only variable assignments
 7:  fcdee92f64c <  -:  ----------- Makefile: change "ifndef NO_PERL" to "ifdef NO_PERL"
 8:  1e25b532ca2 !  1:  90d1e77ad5e Makefile: adjust Perl-related comments & whitespace
    @@ Metadata
     Author: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
     
      ## Commit message ##
    -    Makefile: adjust Perl-related comments & whitespace
    +    Makefile: remove -DPAGER_ENV from BASIC_CFLAGS
     
    -    Folllow-up my 4070c9e09fc (Makefile: don't re-define PERL_DEFINES,
    -    2021-05-05) and move the rest of the assignments to PERL_DEFINES to
    -    one place.
    +    Remove -DPAGER_ENV from the BASIC_CFLAGS and instead have it passed
    +    via the EXTRA_CPPFLAGS passed when compiling pager.c.
    +
    +    This doesn't change anything except to make it clear that only pager.c
    +    needs this, as it's the only user of this define. See
    +    995bc22d7f8 (pager: move pager-specific setup into the build,
    +    2016-08-04) for the commit that originally added this.
     
         Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
     
      ## Makefile ##
    -@@ Makefile: PERL_DEFINES += $(perllibdir_SQ)
    - PERL_DEFINES += $(RUNTIME_PREFIX)
    - PERL_DEFINES += $(NO_PERL_CPAN_FALLBACKS)
    - PERL_DEFINES += $(NO_GETTEXT)
    -+PERL_DEFINES += $(gitexecdir)
    -+PERL_DEFINES += $(perllibdir)
    -+PERL_DEFINES += $(localedir)
    -+PERL_DEFINES := $(subst $(space),:,$(PERL_DEFINES))
    - 
    --# Support Perl runtime prefix. In this mode, a different header is installed
    --# into Perl scripts.
    - ifdef RUNTIME_PREFIX
    --
    - PERL_HEADER_TEMPLATE = perl/header_templates/runtime_prefix.template.pl
    - 
    --# Don't export a fixed $(localedir) path; it will be resolved by the Perl header
    --# at runtime.
    -+# The RUNTIME_PREFIX header defines $Git::I18N::TEXTDOMAINDIR, so
    -+# $(perl_localedir_SQ) won't be needed
    - perl_localedir_SQ =
    --
    +@@ Makefile: ifdef DEFAULT_HELP_FORMAT
    + BASIC_CFLAGS += -DDEFAULT_HELP_FORMAT='"$(DEFAULT_HELP_FORMAT)"'
      endif
      
    --PERL_DEFINES += $(gitexecdir) $(perllibdir) $(localedir)
    +-PAGER_ENV_SQ = $(subst ','\'',$(PAGER_ENV))
    +-PAGER_ENV_CQ = "$(subst ",\",$(subst \,\\,$(PAGER_ENV)))"
    +-PAGER_ENV_CQ_SQ = $(subst ','\'',$(PAGER_ENV_CQ))
    +-BASIC_CFLAGS += -DPAGER_ENV='$(PAGER_ENV_CQ_SQ)'
     -
    - $(SCRIPT_PERL_GEN): % : %.perl GIT-PERL-DEFINES GIT-PERL-HEADER GIT-VERSION-FILE
    - 	$(QUIET_GEN) \
    - 	sed -e '1{' \
    -@@ Makefile: $(SCRIPT_PERL_GEN): % : %.perl GIT-PERL-DEFINES GIT-PERL-HEADER GIT-VERSION-FILE
    - 	chmod +x $@+ && \
    - 	mv $@+ $@
    + ALL_CFLAGS += $(BASIC_CFLAGS)
    + ALL_LDFLAGS += $(BASIC_LDFLAGS)
    + 
    +@@ Makefile: builtin/help.sp builtin/help.s builtin/help.o: EXTRA_CPPFLAGS = \
    + 	'-DGIT_MAN_PATH="$(mandir_relative_SQ)"' \
    + 	'-DGIT_INFO_PATH="$(infodir_relative_SQ)"'
      
    --PERL_DEFINES := $(subst $(space),:,$(PERL_DEFINES))
    - GIT-PERL-DEFINES: FORCE
    - 	@FLAGS='$(PERL_DEFINES)'; \
    - 	    if test x"$$FLAGS" != x"`cat $@ 2>/dev/null`" ; then \
    ++PAGER_ENV_SQ = $(subst ','\'',$(PAGER_ENV))
    ++PAGER_ENV_CQ = "$(subst ",\",$(subst \,\\,$(PAGER_ENV)))"
    ++PAGER_ENV_CQ_SQ = $(subst ','\'',$(PAGER_ENV_CQ))
    ++pager.sp pager.s pager.o: EXTRA_CPPFLAGS = \
    ++	-DPAGER_ENV='$(PAGER_ENV_CQ_SQ)'
    ++
    + version.sp version.s version.o: GIT-VERSION-FILE GIT-USER-AGENT
    + version.sp version.s version.o: EXTRA_CPPFLAGS = \
    + 	'-DGIT_VERSION="$(GIT_VERSION)"' \
 9:  77d9855bfcf <  -:  ----------- Makefile: correct "GIT-PERL-{DEFINES,HEADER}" dependency graph
10:  6004cdcd8d9 <  -:  ----------- Makefile: create a GIT-PYTHON-DEFINES, like "PERL"
11:  17b30e96057 <  -:  ----------- Makefile: stop needing @@GIT_VERSION@@ in *.perl scripts
12:  30ddf7da2c8 <  -:  ----------- Makefiles: add "shared.mak", move ".DELETE_ON_ERROR" to it
13:  f378a7dc35e <  -:  ----------- Makefile: move $(comma), $(empty) and $(space) to shared.mak
14:  13cbb851d32 <  -:  ----------- Makefile: re-add and use the "shellquote" macros
15:  337953e4994 <  -:  ----------- Makefile: add a "TRACK_template" for GIT-*{FLAGS,DEFINES,...}
16:  5bb597c1993 <  -:  ----------- Makefile: add "$(QUIET)" boilerplate to shared.mak
17:  3c4d0589667 <  -:  ----------- Makefile: use $(wspfx) for $(QUIET...) in shared.mak
18:  be5882b2c99 <  -:  ----------- Makefiles: add and use wildcard "mkdir -p" template
19:  2710f8af6cd <  -:  ----------- Makefile: correct the dependency graph of hook-list.h
20:  59f22a0269a <  -:  ----------- Makefile: use $(file) I/O instead of "FORCE" when possible
21:  dd569a59c74 <  -:  ----------- Makefile: disable GNU make built-in wildcard rules
22:  4168a7e3b30 <  -:  ----------- Makefile: define $(LIB_H) in terms of $(FIND_SOURCE_FILES)
23:  48a3927d972 <  -:  ----------- Makefile: move ".SUFFIXES" rule to shared.mak

 Makefile | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index 9c00a793e47..af0bbfb5d2f 100644
--- a/Makefile
+++ b/Makefile
@@ -2111,11 +2111,6 @@ ifdef DEFAULT_HELP_FORMAT
 BASIC_CFLAGS += -DDEFAULT_HELP_FORMAT='"$(DEFAULT_HELP_FORMAT)"'
 endif
 
-PAGER_ENV_SQ = $(subst ','\'',$(PAGER_ENV))
-PAGER_ENV_CQ = "$(subst ",\",$(subst \,\\,$(PAGER_ENV)))"
-PAGER_ENV_CQ_SQ = $(subst ','\'',$(PAGER_ENV_CQ))
-BASIC_CFLAGS += -DPAGER_ENV='$(PAGER_ENV_CQ_SQ)'
-
 ALL_CFLAGS += $(BASIC_CFLAGS)
 ALL_LDFLAGS += $(BASIC_LDFLAGS)
 
@@ -2230,6 +2225,12 @@ builtin/help.sp builtin/help.s builtin/help.o: EXTRA_CPPFLAGS = \
 	'-DGIT_MAN_PATH="$(mandir_relative_SQ)"' \
 	'-DGIT_INFO_PATH="$(infodir_relative_SQ)"'
 
+PAGER_ENV_SQ = $(subst ','\'',$(PAGER_ENV))
+PAGER_ENV_CQ = "$(subst ",\",$(subst \,\\,$(PAGER_ENV)))"
+PAGER_ENV_CQ_SQ = $(subst ','\'',$(PAGER_ENV_CQ))
+pager.sp pager.s pager.o: EXTRA_CPPFLAGS = \
+	-DPAGER_ENV='$(PAGER_ENV_CQ_SQ)'
+
 version.sp version.s version.o: GIT-VERSION-FILE GIT-USER-AGENT
 version.sp version.s version.o: EXTRA_CPPFLAGS = \
 	'-DGIT_VERSION="$(GIT_VERSION)"' \
-- 
2.34.1.1119.g7a3fc8778ee

